src/Result.php

```diff
@@ -242,4 +243,52 @@ private function ensureHasKeyValue(): void
             throw NoKeyValue::fromColumnCount($columnCount);
         }
     }
+
+    /**
+     * BC layer for a wide-spread use-case of old DBAL APIs
+     *
+     * @deprecated This API is deprecated and will be removed after 2022
+     *
+     * @return array<array-key,mixed>|false
```

Comment:
`array<array-key,…>` looks unnecessarily verbose, it's effectively `array<mixed>`. However, is it even always an array? E.g. `fetch(FetchMode::COLUMN)` will return one value.